lib/components/Tabs/Tab.tsx

```diff
@@ -0,0 +1,252 @@
+    }
+  }, [isSelected, isFocused, scrollContainer, space, paddingX, grid]);
+
+  const onKeyUp = (event: KeyboardEvent<HTMLButtonElement>) => {
+    const targetKey = normalizeKey(event);
+
+    if (targetKey === 'Tab') {
+      return;
+    }
+
+    if (onChange && (targetKey === 'Enter' || targetKey === ' ')) {
+      onChange(tabListItemIndex, item);
+    }
+
+    const action: Record<string, Action> = {
+      ArrowRight: { type: TAB_BUTTON_RIGHT },
+      ArrowLeft: { type: TAB_BUTTON_LEFT },
+      Home: { type: TAB_BUTTON_HOME },
+      End: { type: TAB_BUTTON_END },
+      Enter: { type: TAB_BUTTON_ENTER, value: tabListItemIndex },
+      ' ': { type: TAB_BUTTON_SPACE, value: tabListItemIndex },
+    };
+
+    if (action[targetKey]) {
+      dispatch(action[targetKey]);
+    }
+  };
+
+  const onKeyDown = (event: KeyboardEvent<HTMLButtonElement>) => {
+    const targetKey = normalizeKey(event);
+
+    if (targetKey === 'Tab' && focusedTabIndex !== null) {
+      dispatch({ type: TAB_BUTTON_TAB });
+      return;
+    }
+
+    // Prevent arrow keys scrolling the document while navigating the tabs
+    const isArrowPress = targetKey.indexOf('Arrow') === 0;
+    // Prevent enter or space press from triggering the click handler
+    const isActionKeyPress = targetKey === 'Enter' || targetKey === ' ';
+
+    if (isArrowPress || isActionKeyPress) {
+      event.preventDefault();
+    }
+  };
+
+  const tabTextSize = 'standard';
+
+  return (
+    <Box
+      component="button"
+      {...a11y.tabProps({ tabIndex: tabListItemIndex, isSelected })}
+      ref={tabRef}
+      onKeyUp={onKeyUp}
+      onKeyDown={onKeyDown}
+      onClick={(event: MouseEvent) => {
+        event.stopPropagation();
+        event.preventDefault();
+
+        if (onChange) {
+          onChange(tabListItemIndex, item);
+        }
+
+        dispatch({ type: TAB_BUTTON_CLICK, value: tabListItemIndex });
+      }}
+      onFocus={
+        isSelected
+          ? () =>
+              dispatch({
+                type: TAB_LIST_FOCUSED,
+                value: tabListItemIndex,
+              })
+          : undefined
+      }
+      display="flex"
+      alignItems="center"
+      textAlign="left"
+      borderRadius="standard"
+      cursor="pointer"
+      outline="none"
+      position="relative"
+      paddingX={paddingX}
+      paddingY="medium"
+      className={[
+        styles.tab,
+        tabListItemIndex > 0 ? styles.hairlineMarginLeft : null,
+      ]}
+      {...buildDataAttributes(data)}
+    >
+      {/*
+        Rendering Text component to provide rendering context
+        for both icons and text labels
+      */}
+      <Text
+        {...a11y.tabLabelProps({ tabIndex: tabListItemIndex })}
+        size={tabTextSize}
+        weight="medium"
+        align="center"
+        tone={isSelected ? 'formAccent' : 'secondary'}
+      >
+        {children}
+      </Text>
+      {badge ? <Box paddingLeft="xsmall">{badge}</Box> : undefined}
```

Comment:
Noticed the badge adds overall height to the `Tabs`, not sure if it would be better/desired to use a negative margin to avoid this?

Reply:
Fixed.

Reply:
Nice catch, by the way.